zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -206,6 +212,8 @@ pub enum GreedyInputSelectorError<ChangeStrategyErrT, NoteRefT> {
     Balance(BalanceError),
     /// A unified address did not contain a supported receiver.
     UnsupportedAddress(Box<UnifiedAddress>),
+    /// Support for transparent-source-only addresses requires the transparent-inputs feature.
+    UnsupportedTransparentSourceOnlyAddress,
```

Comment:
Breaking change, will require a changelog entry. We could shoehorn this into another error if desired.